### PR TITLE
feat(ci): bring back ci

### PR DESCRIPTION
**Pull request by jacobowitz**
WIP, used to check status of CI

This replaces the temporarily testing for StarRouting with the regular CI workflow. JinaD related tests in `distributed` and `daemon` will be disabled though and activated later.

- [x] Check Flows with inspect pods 
- [x] Disable daemon/distributed tests
- [x] Fix issues integration test
- [x] Fix rolling update integration test
- [x] Fix scale integration test
- [x] Fix latency tracking
- [x] Implement target_peapod and make test_func_routing work

### CODE REVIEW

**Comment by jacobowitz** [comment]
rebased against feat-star-routing after merging the flow PR

File: jina/helloworld/multimodal/app.py [hunk 1]
```diff
@@ -63,10 +63,8 @@ def hello_world(args):
         f.index(inputs=from_csv(fp), request_size=10, show_progress=True)
 
     # search it!
+    os.environ['PORT_EXPOSE'] = str(args.port_expose)
```
Comment (JoanFM) [inline jina/helloworld/multimodal/app.py:66]: why this change, if the `Flow` did not enter its context it is still safe to adapt

File: jina/peapods/peas/container.py [hunk 2]
```diff
@@ -188,8 +188,8 @@ def run(
 
     logger = JinaLogger(name, **vars(args))
 
-    cancel = asyncio.Event()
-    fail_to_start = asyncio.Event()
+    cancel = threading.Event()
```
Comment (JoanFM) [inline jina/peapods/peas/container.py:191]: why? it does not really matter right?

File: jina/peapods/pods/__init__.py [hunk 3]
```diff
@@ -331,6 +334,17 @@ async def scale(self, replicas: int):
                 )  # scale down has some challenges with the exit fifo
             self.pod_args.replicas = replicas
 
+        def has_forked_processes(self):
```
Comment (JoanFM) [inline jina/peapods/pods/__init__.py:337]: make it a private property I'd say

File: jina/peapods/runtimes/gateway/request_handling.py [hunk 4]
```diff
@@ -32,15 +32,25 @@ def _handle_request(request: 'Request') -> 'asyncio.Future':
         r = request.routes.add()
         r.pod = 'gateway'
         r.start_time.GetCurrentTime()
-        for origin_node in request_graph.origin_nodes:
-            leaf_tasks = origin_node.get_leaf_tasks(
-                connection_pool, Message(None, request), None
+        # If the request is targeting a specific pod, we can send directly to the pod instead of querying the graph
+        if request.header.target_peapod:
```
Comment (JoanFM) [inline jina/peapods/runtimes/gateway/request_handling.py:36]: Considere this as a love emoji being sent

File: jina/peapods/runtimes/worker/__init__.py [hunk 5]
```diff
@@ -62,7 +62,9 @@ async def async_cancel(self):
         """Stop the GRPC server"""
         self.logger.debug('Cancel WorkerRuntime')
 
-        await self._grpc_server.stop(0)
+        # 0.5 gives the runtime some time to complete outstanding responses
+        # this should be handled better, 0.5 is a rather random number
```
Comment (JoanFM) [inline jina/peapods/runtimes/worker/__init__.py:66]: We have to keep this in mind for the robustness work later on

File: tests/integration/gateway_clients/test_streaming.py [hunk 6]
```diff
@@ -177,17 +177,6 @@ def create_runtime(graph_dict: Dict, protocol: str, port_in: int, prefetch: int
     'protocol, inputs',
     [
         ('grpc', slow_async_gen),
-        pytest.param(
```
Comment (JoanFM) [inline tests/integration/gateway_clients/test_streaming.py:180]: why did u remove all this? It should be added back

File: tests/integration/gateway_clients/test_streaming.py [hunk 7]
```diff
@@ -93,7 +93,7 @@ async def _compute_response():
 
         async def task_wrapper():
             response_msg = await _compute_response()
-            return [response_msg]
```
Comment (JoanFM) [inline tests/integration/gateway_clients/test_streaming.py:96]: Was this failing?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/peapods/peas/container.py (hunk 2 above)
Comment (jacobowitz) [reply jina/peapods/peas/container.py:191]: it does, an `asyncio.Event()` will fail if there is no asyncio loop running, which is not guranteed at this point. The `rolling_update` tests were failing due to this.

File: tests/integration/gateway_clients/test_streaming.py (hunk 7 above)
Comment (jacobowitz) [reply tests/integration/gateway_clients/test_streaming.py:96]: yes, the interface is wrong, a call returns only a single message. I think that changed at some point in the last weeks and was not changed here.

File: tests/integration/gateway_clients/test_streaming.py (hunk 6 above)
Comment (jacobowitz) [reply tests/integration/gateway_clients/test_streaming.py:180]: true, probably I debugged sth and forgot to readd, will do

File: jina/peapods/pods/__init__.py (hunk 3 above)
Comment (jacobowitz) [reply jina/peapods/pods/__init__.py:337]: hmm why private? The Pod needs to use it

**Comment by jacobowitz** [comment]
squashed commits because I had some malformed commit messages
